modules/nw-how-nw-iface-selected.adoc

```diff
@@ -0,0 +1,78 @@
+// Applies to 4.6 and newer.
+
+:ign-config-version: 3.1.0
+ifeval::[{product-version} > 4.6]
+:ign-config-version: 3.2.0
+endif::[]
+
+[id="nw-how-nw-iface-selected_{context}"]
+= How the network interface is selected
+
+For deployments on bare metal or with virtual machines that have more than one network interface controller (NIC), the NIC that {product-title} uses for communication with the Kubernetes API server is determined by a script that runs as a systemd unit.
+The `nodeip-configuration.service` systemd unit runs when the node boots.
+The script iterates through the network interfaces on the node and the first network interface that is configured with a subnet than can host the IP address for the API server is selected for {product-title} communication.
+
+After the script determines the correct NIC, the script creates the `/etc/systemd/system/kubelet.service.d/20-nodenet.conf` file.
+The `20-nodenet.conf` file sets the `KUBELET_NODE_IP` environment variable to the IP address that was selected.
+
+When the kubelet service starts, it reads the value of the environment variable from the `20-nodenet.conf` file and sets the IP address as the value to the `--node-ip` kubelet command-line argument.
+As a result, the kubelet service uses the selected IP address as the node IP address.
+
+If hardware or networking is reconfigured after installation, it is possible that the script for the `nodeip-configuration.service` unit can select a different NIC after a reboot and prevent {product-title} communication.
+
+If {product-title} communication is disrupted because a different NIC is selected, one strategy for overriding the selection process is to set the correct IP address explicitly.
+The following list identifies the high-level steps and considerations:
+
+* Create a shell script that determines the IP address to use for {product-title} communication. Have the script create a configuration file such as `/etc/systemd/system/kubelet.service.d/98-nodenet-override.conf` and set the `KUBELET_NODE_IP` environment variable to the IP address.
+
+* Create a machine config object with the shell script as a base64-encoded string and use the Machine Config Operator to deploy the script to the nodes at a file system path such as `/usr/local/bin/override-node-ip.sh`.
+
+* Do not overwrite the `/etc/systemd/system/kubelet.service.d/20-nodenet.conf` file. Specify a file name with a lexicographically higher value such as `98-nodenet-override.conf` in the same directory path. The goal is to have the custom systemd unit file run after `20-nodenet.conf` and override the value of the environment variable.
+
```

Comment:
Can/should you combine the first and third bullets? The information about the _lexicographically higher value_ seems relevant to bullet 1.
I wonder, too, if _lexicographically_ will translate well ??  _Specify a file name with a higher numeric value..._

Reply:
yes, I agree, if the first steps already said the shell script `usr/local/bin/override-node-ip.sh` will generate the config file `/etc/systemd/system/kubelet.service.d/98-nodenet-override.conf`.  thus the third steps should be redundant

Reply:
I revised and reordered them so the original third bullet is now the second bullet.  My instinct is to be cautious and explain it gradually.

Reply:
```
* Create a shell script that determines the IP address to use for {product-title} communication. Have the script create a custom unit file such as `/etc/systemd/system/kubelet.service.d/98-nodenet-override.conf`. Use the custom unit file, `98-nodenet-override.conf`, to set the `KUBELET_NODE_IP` environment variable to the IP address.

* Do not overwrite the `/etc/systemd/system/kubelet.service.d/20-nodenet.conf` file. Specify a file name with a numerically higher value such as `98-nodenet-override.conf` in the same directory path. The goal is to have the custom unit file run after `20-nodenet.conf` and override the value of the environment variable.
```
first step already create file `98-nodenet-override.conf`. why the second step still need?